File: 백준/Silver/2828. 사과 담기 게임/사과 담기 게임.cc
#include <bits/stdc++.h>
using namespace std;

int N, M, J, pos, ret;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    cin >> N >> M >> J;

    int l = 1, r = 1;
    for (int i=0; i<J; i++) {
        r = l + M - 1;

        cin >> pos;
        if (l <= pos && pos <= r) continue;
        else {
            if (pos < l) {
                ret += (l - pos);
                l = pos;
            } else {
                ret += (pos - r);
                l += pos - r;
            }
        }
    }
    cout << ret << '\n';
    return 0;
}

File: 백준/Silver/2828. 사과 담기 게임/사과 담기 게임.py
N, M = map(int, input().split())
J = int(input())

ret = 0
l, r = 1, 1

for _ in range(J):
    r = l + M - 1

    pos = int(input())
    if l <= pos <= r:
        continue

    if pos < l:
        ret += (l - pos)
        l = pos
    elif pos > r:
        ret += (pos - r)
        l += (pos - r)

print(ret)